### PR TITLE
Creation of components through MLContext and cleanup (text related transforms)

### PR DESCRIPTION
**Pull request by abgoswam**
Towards #1798 , #1758, #1760

The following transform estimators are being addressed:
- LatentDirichletAllocationEstimator
- WordEmbeddingsExtractingEstimator
- TokenizingByCharactersEstimator
- WordTokenizingEstimator
- WordBagEstimator
- WordHashBagEstimator
- NgramExtractingEstimator
- NgramHashingEstimator
- StopWordsRemovingEstimator
- CustomStopWordsRemovingEstimator
- TextNormalizingEstimator

The changes are as follows :
1. Internalize constructors of estimators and transformers #1798
2. Rename `Arguments` -> `Options` #1798
3. Internalize `Options` when they are not used by public constructor. #1758 
4. Rename `Options` objects as `options` (instead of `args` or `advancedSettings` used so far) #1798
5. Move ColumnInfo to the estimators #1760

### CODE REVIEW

**Comment by artidoro** [comment]
>         public sealed class Column : OneToOneColumn

can you make this internal as well?


---
Refers to: src/Microsoft.ML.Transforms/Text/LdaTransform.cs:109 in 92e6f2d. [](commit_id = 92e6f2da0bc1f9e1ba4dc7fb4009219e4435f361, deletion_comment = False)

**Comment by artidoro** [comment]
>         public sealed class LdaSummary

If possible this one too should become internal. I am not sure where it is used though, so might need to double check.

---
Refers to: src/Microsoft.ML.Transforms/Text/LdaTransform.cs:167 in 92e6f2d. [](commit_id = 92e6f2da0bc1f9e1ba4dc7fb4009219e4435f361, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/LdaTransform.cs [hunk 1]
```diff
@@ -1134,21 +965,190 @@ internal static class Defaults
...
+        internal LatentDirichletAllocationEstimator(IHostEnvironment env, params ColumnInfo[] columns)
         {
             Contracts.CheckValue(env, nameof(env));
             _host = env.Register(nameof(LatentDirichletAllocationEstimator));
             _columns = columns.ToImmutableArray();
         }
 
+        public sealed class ColumnInfo
+        {
+            public readonly string Name;
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: >public [](start = 12, length = 6)

Could you add a summary tag for each one of these public entries?

And could you add a summary for the ColumnInfo object?

**Comment by abgoswam** [comment]
>         public sealed class LdaSummary

this one is currently used by Static API  via the OnFit()  delegate,   to provide details about the topics discovered by LightLDA.

making it internal would  break the Static API 

---
In reply to: [460088862](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460088862) [](ancestors = 460088862)

---
Refers to: src/Microsoft.ML.Transforms/Text/LdaTransform.cs:167 in 92e6f2d. [](commit_id = 92e6f2da0bc1f9e1ba4dc7fb4009219e4435f361, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
>         /// </summary>

nit: I would end summary here, and wrap next paragraph into <remark>

---
Refers to: src/Microsoft.ML.Transforms/Text/TextCatalog.cs:483 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
>         public override SchemaShape GetOutputSchema(SchemaShape inputSchema)

```        /// <summary>
        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
        /// Used for schema propagation and verification in a pipeline.
        /// </summary>```

---
Refers to: src/Microsoft.ML.Transforms/Text/TokenizingByCharacters.cs:587 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ internal bool TryUnparse(StringBuilder sb)
         /// </summary>
         public sealed class TokenizeColumn : OneToOneColumn { }
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: >public [](start = 8, length = 6)

also internal

**Comment by Ivanidzo4ka** [comment]
>     }

internal

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:41 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs [hunk 3]
```diff
@@ -220,7 +220,7 @@ internal bool TryUnparse(StringBuilder sb)
 
         /// <summary>
         /// This class is a merger of <see cref="ValueToKeyMappingTransformer.Options"/> and
-        /// <see cref="NgramExtractingTransformer.Arguments"/>, with the allLength option removed.
+        /// <see cref="NgramExtractingTransformer.Options"/>, with the allLength option removed.
         /// </summary>
         public abstract class ArgumentsBase
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: >public [](start = 8, length = 6)

internal?

**Comment by Ivanidzo4ka** [comment]
>         public static IDataTransform Create(IHostEnvironment env, NgramExtractorArguments extractorArgs, IDataView input,

can you merge with master, I think I tackle it recently.

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:369 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs [hunk 4]
```diff
@@ -850,6 +833,27 @@ internal WordEmbeddingsExtractingEstimator(IHostEnvironment env, string customMo
...
 
+        /// <summary>
+        /// Information for each column pair.
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            public readonly string Name;
+            public readonly string InputColumnName;
+
+            public ColumnInfo(string name, string inputColumnName = null)
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs]: summary to constructor.

**Comment by Ivanidzo4ka** [comment]
>         public class Column : OneToOneColumn

internal

---
Refers to: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs:43 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by artidoro** [comment]
>         public sealed class LdaSummary

Ok sounds good thanks for explaining!

---
In reply to: [460757704](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460757704) [](ancestors = 460757704,460088862)

---
Refers to: src/Microsoft.ML.Transforms/Text/LdaTransform.cs:167 in 92e6f2d. [](commit_id = 92e6f2da0bc1f9e1ba4dc7fb4009219e4435f361, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordHashBagProducingTransform.cs [hunk 5]
```diff
@@ -296,15 +296,15 @@ internal static class DefaultArguments
 
         [TlcModule.Component(Name = "NGramHash", FriendlyName = "NGram Hash Extractor Transform", Alias = "NGramHashExtractorTransform,NGramHashExtractor",
                             Desc = "Extracts NGrams from text and convert them to vector using hashing trick.")]
-        public sealed class NgramHashExtractorArguments : ArgumentsBase, INgramExtractorFactoryFactory
+        internal sealed class NgramHashExtractorArguments : ArgumentsBase, INgramExtractorFactoryFactory
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/Text/WordHashBagProducingTransform.cs]: >NgramHashExtractorArguments [](start = 30, length = 27)

Same here, NgramHashExtractorOptions?

File: src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs [hunk 6]
```diff
@@ -245,7 +245,7 @@ public sealed class WordHashBagEstimator : TrainedWrapperEstimatorBase
         public override TransformWrapper Fit(IDataView input)
         {
             // Create arguments.
-            var args = new WordHashBagProducingTransformer.Arguments
+            var args = new WordHashBagProducingTransformer.Options
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs]: >args [](start = 16, length = 4)

Could you rename to options?

File: src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs [hunk 7]
```diff
@@ -108,7 +108,7 @@ public sealed class WordBagEstimator : TrainedWrapperEstimatorBase
         public override TransformWrapper Fit(IDataView input)
         {
             // Create arguments.
-            var args = new WordBagBuildingTransformer.Arguments
+            var args = new WordBagBuildingTransformer.Options
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs]: >args [](start = 16, length = 4)

same here, could you rename to options?

**Comment by artidoro** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) =>

Can you make this internal?

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:551 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) => (type.GetItemType() is TextType);

Can we make this one internal too?

---
Refers to: src/Microsoft.ML.Transforms/Text/TextNormalizing.cs:453 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) => type.GetItemType() is TextType;

Same here, can we make this internal?

---
Refers to: src/Microsoft.ML.Transforms/Text/TokenizingByCharacters.cs:557 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         {

/// <summary>
/// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
/// Used for schema propagation and verification in a pipeline.
/// </summary>

---
Refers to: src/Microsoft.ML.Transforms/Text/TextNormalizing.cs:501 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         {

/// <summary>
/// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
/// Used for schema propagation and verification in a pipeline.
/// </summary>

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:1077 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         {

/// <summary>
/// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
/// Used for schema propagation and verification in a pipeline.
/// </summary>

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:587 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs (hunk 2 above)
Comment (artidoro) [reply src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: The class is internal, so it should be fine

---
In reply to: [254030739](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254030739) [](ancestors = 254030739)

**Comment by artidoro** [comment]
>         public static VersionInfo GetVersionInfo()

Could you make this internal?

---
Refers to: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs:79 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         }

Since we use this in the estimator for `ColumnInfo`, can you move it there?


---
Refers to: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs:635 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
>         public abstract class ArgumentsBase : TransformInputBase

Can you make this internal too?


---
Refers to: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs:67 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by artidoro** [comment]
> 

Can this be made internal?

---
Refers to: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs:409 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs [hunk 8]
```diff
@@ -460,10 +437,29 @@ public WordTokenizingEstimator(IHostEnvironment env, (string outputColumnName, s
...
+            public ColumnInfo(string name, string inputColumnName = null, char[] separators = null)
+            {
+                Name = name;
+                InputColumnName = inputColumnName ?? name;
+                Separators = separators ?? new[] { ' ' };
+            }
+        }
 
         public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
         {
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/Text/WordTokenizing.cs]: /// <summary>
/// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
/// Used for schema propagation and verification in a pipeline.
/// </summary>

**Comment by abgoswam** [comment]
>         public sealed class NgramExtractorArguments : ArgumentsBase, INgramExtractorFactoryFactory

I will create a separate issue to scrub all the entrypoint APIs and fix their namings. If you see they derive from `ArgumentsBase`  so need to consider if they should have suffix *Options

---
In reply to: [460829905](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460829905) [](ancestors = 460829905)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:249 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs (hunk 2 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: deleted actually. had 0 references :)

---
In reply to: [254030739](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254030739) [](ancestors = 254030739)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs (hunk 3 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: fixed

---
In reply to: [254030994](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254030994) [](ancestors = 254030994)

File: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs (hunk 4 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs]: fixed

---
In reply to: [254031689](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254031689) [](ancestors = 254031689)

**Comment by abgoswam** [comment]
>         public override SchemaShape GetOutputSchema(SchemaShape inputSchema)

fixed

---
In reply to: [460788078](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460788078) [](ancestors = 460788078)

---
Refers to: src/Microsoft.ML.Transforms/Text/TokenizingByCharacters.cs:587 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by abgoswam** [comment]
>     }

fixed i think ?

---
In reply to: [460788328](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460788328) [](ancestors = 460788328)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:41 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public static IDataTransform Create(IHostEnvironment env, NgramExtractorArguments extractorArgs, IDataView input,

fixed. looks like u had missed it.  ;)

---
In reply to: [460788737](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460788737) [](ancestors = 460788737)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:369 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public class Column : OneToOneColumn

fixed

---
In reply to: [460789322](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460789322) [](ancestors = 460789322)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs:43 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public void Dispose()

implementing IDisposable, so cannot make this internal.  

---
In reply to: [460827459](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460827459) [](ancestors = 460827459)

---
Refers to: src/Microsoft.ML.Transforms/Text/LdaTransform.cs:684 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs (hunk 6 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs]: fixed

---
In reply to: [254075307](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254075307) [](ancestors = 254075307)

File: src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs (hunk 7 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs]: fixed

---
In reply to: [254075392](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254075392) [](ancestors = 254075392)

**Comment by abgoswam** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) =>

fixed

---
In reply to: [460835302](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460835302) [](ancestors = 460835302)

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:551 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) => (type.GetItemType() is TextType);

fixed

---
In reply to: [460836062](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460836062) [](ancestors = 460836062)

---
Refers to: src/Microsoft.ML.Transforms/Text/TextNormalizing.cs:453 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public static bool IsColumnTypeValid(ColumnType type) => type.GetItemType() is TextType;

fixed

---
In reply to: [460836382](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460836382) [](ancestors = 460836382)

---
Refers to: src/Microsoft.ML.Transforms/Text/TokenizingByCharacters.cs:557 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         {

fixed

---
In reply to: [460836546](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460836546) [](ancestors = 460836546)

---
Refers to: src/Microsoft.ML.Transforms/Text/TextNormalizing.cs:501 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         {

fixed

---
In reply to: [460836602](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460836602) [](ancestors = 460836602)

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:1077 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         {

fixed

---
In reply to: [460836637](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460836637) [](ancestors = 460836637)

---
Refers to: src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs:587 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         public static VersionInfo GetVersionInfo()

fixed

---
In reply to: [460837490](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460837490) [](ancestors = 460837490)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs:79 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
>         }

fixed

---
In reply to: [460838030](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460838030) [](ancestors = 460838030)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordEmbeddingsExtractor.cs:635 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

**Comment by abgoswam** [comment]
> 

fixed

---
In reply to: [460838678](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460838678) [](ancestors = 460838678)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs:409 in aae84eb. [](commit_id = aae84ebc932c5c9d5961d71c27c0f701a2a067c7, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/WordTokenizing.cs (hunk 8 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/WordTokenizing.cs]: fixed

---
In reply to: [254082141](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254082141) [](ancestors = 254082141)

File: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs [hunk 9]
```diff
@@ -42,7 +42,7 @@ internal sealed class ExtractorColumn : ManyToOneColumn
 
     internal static class WordBagBuildingTransformer
     {
-        public sealed class Column : ManyToOneColumn
+        internal sealed class Column : ManyToOneColumn
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/WordBagTransform.cs]: >internal [](start = 8, length = 8)

it's already static class, you no longer need to make it subclass internal :)

File: src/Microsoft.ML.Transforms/Text/LdaTransform.cs [hunk 10]
```diff
@@ -1134,23 +965,235 @@ internal static class Defaults
...
+        /// Describes how the transformer handles one column pair.
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            /// <summary>
+            /// Name of the column resulting from the transformation of <cref see="InputColumnName"/>.
+            /// </summary>
+            public readonly string Name;
+            /// <summary>
+            /// Name of column to transform. If set to <see langword="null"/>, the value of the <cref see="Name"/> will be used as source.
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: > /// Name of column to transform. If set to <see langword="null"/>, the value of the <cref see="Name"/> will be used as source. [](start = 11, length = 127)

as Zeeshan A pointed in other PRs, you need to left only first sentence. It's always populated with value.

Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: not quite sure what u mean .

scrub whole PR for what ?

---
In reply to: [254093004](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254093004) [](ancestors = 254093004)

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: If you add anywhere else that comment to InputColumnName inside ColumnInfo, you need to change it.
If you didn't add it in other places, you just need to fix this one.

---
In reply to: [254094414](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254094414) [](ancestors = 254094414,254093004)

**Comment by Ivanidzo4ka** [comment]
>         public static IDataTransform Create(IHostEnvironment env, NgramExtractorArguments extractorArgs, IDataView input,

Well, it's part of internal class now, so I decide it's not worth time to change methods modifiers :)

---
In reply to: [460846657](https://github.com/dotnet/machinelearning/pull/2393#issuecomment-460846657) [](ancestors = 460846657,460788737)

---
Refers to: src/Microsoft.ML.Transforms/Text/WordBagTransform.cs:369 in a81c3d2. [](commit_id = a81c3d23707b61fd97d685e36d6f929deceffb13, deletion_comment = False)

File: src/Microsoft.ML.Transforms/Text/LdaTransform.cs (hunk 10 above)
Comment (abgoswam) [reply src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: So it should just say 

            /// <summary>
            /// Name of column to transform. 
            /// </summary>

?

---
In reply to: [254094755](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254094755) [](ancestors = 254094755,254094414,254093004)

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: @artidoro what do you put in them? Sorry, it fills like billions of same PRs, and it's hard to find for me, but I know what Artidoro address that in his PR recently.

---
In reply to: [254096126](https://github.com/dotnet/machinelearning/pull/2393#discussion_r254096126) [](ancestors = 254096126,254094755,254094414,254093004)

Comment (artidoro) [reply src/Microsoft.ML.Transforms/Text/LdaTransform.cs]: Yes that's how I fixed it!

File: test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs [hunk 11]
```diff
@@ -473,7 +473,7 @@ private void TextFeaturizationOn(string dataPath)
                     BagOfTrichar: r.Message.TokenizeIntoCharacters().ToNgrams(ngramLength: 3, weighting: NgramExtractingEstimator.WeightingCriteria.TfIdf),
 
                     // NLP pipeline 4: word embeddings.
-                    Embeddings: r.Message.NormalizeText().TokenizeText().WordEmbeddings(WordEmbeddingsExtractingTransformer.PretrainedModelKind.GloVeTwitter25D)
+                    Embeddings: r.Message.NormalizeText().TokenizeText().WordEmbeddings(WordEmbeddingsExtractingEstimator.PretrainedModelKind.GloVeTwitter25D)
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs]: >NormalizeText [](start = 42, length = 13)

If you update any of cookbooks you also need to update [https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md](https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md)

I wish we had way to autogenerate it somehow...

**Pull request merged**